hudi-io/src/main/java/org/apache/hudi/io/hfile/HFileRootIndexBlock.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.hudi.io.hfile;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.apache.hudi.io.hfile.ByteUtils.copy;
+import static org.apache.hudi.io.hfile.ByteUtils.decodeVLongSizeOnDisk;
+import static org.apache.hudi.io.hfile.ByteUtils.readInt;
+import static org.apache.hudi.io.hfile.ByteUtils.readLong;
+import static org.apache.hudi.io.hfile.ByteUtils.readVLong;
+
+/**
+ * Represents a {@link HFileBlockType#ROOT_INDEX} block in the "Load-on-open" section.
```

Comment:
Revisit all comments to make sure we don't have some things out of context

Reply:
Comments are polished.